cdc/cdc/kv/region_worker.go

```diff
@@ -730,39 +731,45 @@ func (w *regionWorker) handleResolvedTs(
 		return nil
 	}
 	regionID := state.sri.verID.GetID()
-	// Send resolved ts update in non blocking way, since we can re-query real
-	// resolved ts from region state even if resolved ts update is discarded.
-	// NOTICE: We send any regionTsInfo to resolveLock thread to give us a chance to trigger resolveLock logic
-	// (1) if it is a fallback resolvedTs event, it will be discarded and accumulate penalty on the progress;
-	// (2) if it is a normal one, update rtsManager and check sinceLastResolvedTs
-	select {
-	case w.rtsUpdateCh <- &regionTsInfo{regionID: regionID, ts: newResolvedTsItem(resolvedTs)}:
-	default:
-	}
 
-	if resolvedTs < state.lastResolvedTs {
+	lastResolvedTs0, lastResolvedTs := state.lastResolvedTs[0], state.lastResolvedTs[1]
+	if resolvedTs < lastResolvedTs {
+		// TODO: panic ?
 		log.Warn("The resolvedTs is fallen back in kvclient",
 			zap.String("Event Type", "RESOLVED"),
 			zap.Uint64("resolvedTs", resolvedTs),
-			zap.Uint64("lastResolvedTs", state.lastResolvedTs),
+			zap.Uint64("lastResolvedTs", lastResolvedTs),
 			zap.Uint64("regionID", regionID))
 		return nil
 	}
-	state.lastResolvedTs = resolvedTs
-	// emit a checkpointTs
-	revent := model.RegionFeedEvent{
-		RegionID: regionID,
-		Resolved: &model.ResolvedSpan{
-			Span:       state.sri.span,
-			ResolvedTs: resolvedTs,
-		},
-	}
+	state.lastResolvedTs[0], state.lastResolvedTs[1] = lastResolvedTs, resolvedTs
 
-	select {
-	case w.outputCh <- revent:
-		w.metrics.metricSendEventResolvedCounter.Inc()
-	case <-ctx.Done():
-		return errors.Trace(ctx.Err())
+	if lastResolvedTs0 != lastResolvedTs {
+		// Send resolved ts update in non blocking way, since we can re-query real
+		// resolved ts from region state even if resolved ts update is discarded.
+		// NOTICE: We send any regionTsInfo to resolveLock thread to give us a chance to trigger resolveLock logic
+		// (1) if it is a fallback resolvedTs event, it will be discarded and accumulate penalty on the progress;
+		// (2) if it is a normal one, update rtsManager and check sinceLastResolvedTs
+		select {
+		case w.rtsUpdateCh <- &regionTsInfo{regionID: regionID, ts: newResolvedTsItem(lastResolvedTs)}:
```

Comment:
Suggest to keep the original codes, that push the incoming resolvedTs to `w.rtsUpdateCh`. Just as the comments says, a fall back `resolved-ts` will accumulate penalty. We can also utility it.